doc/nrf/ug_zigbee_adding_clusters.rst

```diff
@@ -0,0 +1,506 @@
+Altering the attribute does nothing more than changing values of a specific variable.
+The application is supposed to react to these changes and produce appropriate behavior, but we need to inform it about these changes first.
+
+To inform the application about attributes changes, you can pass ZCL events to it with a callback that follows generic callback definition (referred to as *ZCL callback*).
+This is shown in the following snippet:
+
+.. code-block:: C++
+
+   typedef void (zb_callback_t)(zb_uint8_t param);
+
+The ``param`` argument passed to the callback contains information about the changed attributes.
+This argument is actually a ZBOSS buffer that contains the `zb_zcl_device_callback_param_t`_ structure whose definition fragment is as follows:
+
+.. code-block:: C++
+
+        /* For the full definition please refer to zboss_api_zcl.h */
+        typedef struct zb_zcl_device_callback_param_s
+        {
+                /** Type of device callback */
+                zb_zcl_device_callback_id_t device_cb_id;
+                zb_uint8_t endpoint;
+                zb_zcl_attr_access_t attr_type;
+
+                /** Return status (see zb_ret_t) */
+                zb_ret_t status;
+
+                /** Callback custom data */
+                union
+                {
+                        zb_zcl_set_attr_value_param_t  set_attr_value_param;
+                        #if defined (ZB_ZCL_SUPPORT_CLUSTER_ON_OFF)
+                        /* Off with effect command, On/Off cluster */
+                        zb_zcl_on_off_set_effect_value_param_t  on_off_set_effect_value_param;
+                        /* */
+                        #endif
+                        #if defined(ZB_ZCL_SUPPORT_CLUSTER_IDENTIFY)
+                        zb_zcl_identify_effect_value_param_t  identify_effect_value_param;
+                        #endif
+
+                        /* .
+                           .
+                           .
+                        */
+
+                        zb_zcl_device_cmd_generic_param_t gnr;
+                } cb_param;
+        } zb_zcl_device_callback_param_t;
+
+Both the On/Off Switch device and the Temperature Sensor device have client roles.
+For this reason, their attributes are not supposed to change.
+To show how you can use the ZCL event, read the following section.
+
+Reading device callback parameters from a ZBOSS buffer
+------------------------------------------------------
+
+The ZCL callback implemented in the :ref:`lib_zigbee_fota` library is a good example of how to use the ZCL event.
+
+To read the device callback parameters from a ZBOSS buffer, complete the following steps:
+
+1. In the :file:`main.c` file of the copied template sample, extend the `zb_zcl_device_callback_param_t`_ structure that contains the ZCL callback parameters in the following manner:
```

Comment:
The structure is some kind of a placeholder for parameters passed to callback.

I propose following sentence:
In the :file:`main.c` file of the copied template sample, use the `zb_zcl_device_callback_param_t`_ structure to get the ZCL callback parameters from the buffer in the following manner: